lib/rex/post/meterpreter/extensions/stdapi/railgun/library.rb

```diff
@@ -317,9 +317,19 @@ def build_response(packet, function, layouts, arch)
       when 'VOID'
         return_hash['return'] = nil
       when 'PCHAR'
-        return_hash['return'] = rec_return_value == 0 ? nil : read_string(rec_return_value)
+        return_hash['return'] = rec_return_value == 0 ? nil : client.railgun.util.read_string(rec_return_value)
+        return_hash['&return'] = rec_return_value
       when 'PWCHAR'
-        return_hash['return'] = rec_return_value == 0 ? nil : read_wstring(rec_return_value)
+        return_hash['return'] = rec_return_value == 0 ? nil : client.railgun.util.read_wstring(rec_return_value)
+        return_hash['&return'] = rec_return_value
+      when 'PULONG_PTR'
+        if client.native_arch == ARCH_X64
+          return_hash['return'] = rec_return_value == 0 ? nil : client.railgun.util.memread(rec_return_value, 8).unpack1('Q')
```

Comment:
This is dangerous: These `memread` calls could return `nil` if they fail to read data, and `.unpack1` will not exist on a `nil` object.

Reply:
Might be an idea to `&.unpack1` here to fix this and so we return `nil` if `memread` returns `nil`.

Reply:
Per discussion this should be `Q<` to specify little endian explicitly.